From: Gordon Malm <bugs-gentoo-org-02@bumpin.org>

Fix a recursive lock -- call to capable() within ptrace_attach().

This patch is present in upstream grsecurity patches as of
grsecurity-2.1.11-2.6.24.3-200803131725.patch.  This patch can
be dropped for any hardened-sources-2.6.24 based on 
grsecurity-2.1.11-2.6.24.3-200803131725.patch or later.

Acked-by: Kerin Millar <kerframil@gmail.com>

--- a/kernel/ptrace.c
+++ b/kernel/ptrace.c
@@ -203,7 +203,7 @@ repeat:
 	/* Go */
 	task->ptrace |= PT_PTRACED | ((task->real_parent != current)
 				      ? PT_ATTACHED : 0);
-	if (capable(CAP_SYS_PTRACE))
+	if (capable_nolog(CAP_SYS_PTRACE))
 		task->ptrace |= PT_PTRACE_CAP;
 
 	__ptrace_link(task, current);
